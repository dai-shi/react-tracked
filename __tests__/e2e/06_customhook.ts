/* global page */

jest.setTimeout(15 * 1000);

describe('06_customhook', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > #app > div:nth-child(1) > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(1) > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(1) > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(1) > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(1) > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(1) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(1) > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(1) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();
  });
});
