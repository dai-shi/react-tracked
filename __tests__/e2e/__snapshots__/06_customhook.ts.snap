// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`06_customhook should work with recorded events 1`] = `
"
    <div id=\\"app\\"><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 2`] = `
"
    <div id=\\"app\\"><div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 3`] = `
"
    <div id=\\"app\\"><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 4`] = `
"
    <div id=\\"app\\"><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 5`] = `
"
    <div id=\\"app\\"><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 6`] = `
"
    <div id=\\"app\\"><div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 7`] = `
"
    <div id=\\"app\\"><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`06_customhook should work with recorded events 8`] = `
"
    <div id=\\"app\\"><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div>
  <script src=\\"/main.js\\"></script>

"
`;
